lib/nerve/reporter/zookeeper.rb

```diff
@@ -158,20 +164,38 @@ def zk_create
       statsd.time('nerve.reporter.zk.create.elapsed_time', tags: ["zk_cluster:#{@zk_cluster}", "zk_path:#{@zk_path}"]) do
         @zk.mkdir_p(@zk_path) unless @zk.exists?(@zk_path)
         @full_key = @zk.create(@key_prefix, :data => @data, :mode => @mode)
+        log.info "nerve: wrote new ZK node of type #{@mode} at #{@full_key}"
       end
     end
 
-    def zk_save
-      return zk_create unless @full_key
+    def zk_save(exists)
+      return zk_create unless exists
 
       begin
         statsd.time('nerve.reporter.zk.save.elapsed_time', tags: ["zk_cluster:#{@zk_cluster}"]) do
           @zk.set(@full_key, @data)
+          log.info "nerve: set data on #{@full_key}"
         end
       rescue ZK::Exceptions::NoNode
         zk_create
       end
     end
+
+    # Renew TTL on the node by "touching" it. If the node exists, it will be
+    # written to with the same data. If it does not exist, it will be
+    # automatically be created.
+    # Note: ephemeral nodes are not written because they are managed by
+    # the session.
+    def renew_ttl(node_stat)
+      return if @node_ttl.nil?
+
+      node_exists = node_stat.nil? || node_stat.exists?
```

Comment:
should it be &&?

Reply:
Good catch! I think it should be `!node_stat.nil? && node_stat.exists?`.